File: leetcode/Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int> &height) {
        int begin = 0, end = height.size() - 1;
        int max_area = 0;
        while (begin <= end && begin < height.size() && end >= 0) {
            max_area = std::max(max_area, std::min(height[begin], height[end]) * (end - begin));
            // come closer
            if (height[end] > height[begin]) {
                ++ begin;
            }
            else {
                -- end;
            }
        }
        return max_area;
    }
};

File: answers/leetcode/Container With Most Water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """

        left = 0
        right = len(height) - 1
        max_area = 0
        while left < right:
            water = min(height[left], height[right]) * (right - left)
            max_area = max(max_area, water)
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return max_area
